sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercionBase.scala

```diff
@@ -0,0 +1,388 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.analysis
+
+import scala.annotation.tailrec
+import scala.collection.mutable
+
+import org.apache.spark.sql.catalyst.expressions.{
+  Alias,
+  CaseWhen,
+  Cast,
+  Concat,
+  Elt,
+  Expression,
+  MapZipWith,
+  Stack,
+  WindowSpecDefinition
+}
+import org.apache.spark.sql.catalyst.plans.logical.{
+  Call,
+  Except,
+  Intersect,
+  LogicalPlan,
+  Project,
+  Union,
+  Unpivot
+}
+import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.connector.catalog.procedures.BoundProcedure
+import org.apache.spark.sql.types.DataType
+
+abstract class TypeCoercionBase extends TypeCoercionHelper {
```

Comment:
to confirm, do we just move it from `TypeCoercion.scala` to here without any change?

Reply:
Exactly